In [1]:
import pandas as pd

In [2]:
Mike_Campbell = pd.read_csv(r"C:\Users\Adrien\Documents\datascience\Projet Prenom\Prenoms.csv",encoding='latin-1', sep=';')

In [3]:
Mike_Campbell

01_prenom 02_genre        03_langage  04_fréquence
0        aaliyah        f  english (modern)           0.0
1         aapeli        m           finnish           0.0
2           aapo        m           finnish           0.0
3          aaren      m,f           english           0.0
4          aarne        m           finnish           0.0
...          ...      ...               ...           ...
11622        zvi        m            jewish           0.1
11623   zvonimir        m          croatian           0.0
11624  zvonimira        f          croatian           0.0
11625     zvonko        m          croatian           0.0
11626    zygmunt        m            polish           0.0

[11627 rows x 4 columns]

In [4]:
df = pd.read_csv(r"C:\Users\Adrien\Documents\datascience\Projet Prenom\dpt2018.csv", sep=';')

In [5]:
df

sexe preusuel annais dpt  nombre
0           1        A   XXXX  XX      27
1           1    AADAM   XXXX  XX      27
2           1    AADEL   XXXX  XX      55
3           1    AADIL   1983  84       3
4           1    AADIL   1992  92       3
...       ...      ...    ...  ..     ...
3624989     2      ÉYA   2014  69       3
3624990     2      ÉYA   XXXX  XX      23
3624991     2    ÏNAYA   XXXX  XX      21
3624992     2     ÖZGE   XXXX  XX      30
3624993     2    ÖZLEM   XXXX  XX      95

[3624994 rows x 5 columns]

Quelle est la proportion totale Femmes / Hommes au cours du temps depuis 1900 ?

In [6]:
nbr_annee = df.groupby(["annais", "sexe"])["nombre"].sum().reset_index()

In [7]:
totaux_par_annee = nbr_annee.groupby("annais")["nombre"].sum().reset_index()
totaux_par_annee.rename(columns={"nombre": "total_naissances"}, inplace=True)

In [8]:
nbr_annee = nbr_annee.merge(totaux_par_annee, on="annais")

In [9]:
nbr_annee['nombre'] = nbr_annee['nombre'] / nbr_annee['total_naissances']

In [10]:
nbr_annee

annais  sexe    nombre  total_naissances
0     1900     1  0.427855            393432
1     1900     2  0.572145            393432
2     1901     1  0.432467            431732
3     1901     2  0.567533            431732
4     1902     1  0.439457            443886
..     ...   ...       ...               ...
235   2017     2  0.473617            573677
236   2018     1  0.527286            564259
237   2018     2  0.472714            564259
238   XXXX     1  0.446006           8261688
239   XXXX     2  0.553994           8261688

[240 rows x 4 columns]

Elle est proche de 50/50

Comment a-t-elle évolué au cours du temps ?

In [11]:
nbr_annee

annais  sexe    nombre  total_naissances
0     1900     1  0.427855            393432
1     1900     2  0.572145            393432
2     1901     1  0.432467            431732
3     1901     2  0.567533            431732
4     1902     1  0.439457            443886
..     ...   ...       ...               ...
235   2017     2  0.473617            573677
236   2018     1  0.527286            564259
237   2018     2  0.472714            564259
238   XXXX     1  0.446006           8261688
239   XXXX     2  0.553994           8261688

[240 rows x 4 columns]

En 1900 il y'avait plus de femmes que d'hommes (43/57) cette donnee a tendance a changer au cours du temps de plus en plus la part d'hommes augumentent et celle des femmes diminue. Jusqu'en 1928 ou les hommes et les femmes sont quasiment sur un pied d'egalité (50/50). Depuis la part d'hommes continuent d'augumenter, jusqu'en 2018 où l'ecart est plus significatif (53/47).

On va maintenant essayer de retrouver (ou non!) quelques uns des résultats présentés
dans l’article du Figaro

Evolution du prénom Marie dans le temps (nbre de naissances de Marie par an) ?

In [12]:
nbr_marie = df.groupby(['annais', 'preusuel'])['nombre'].sum().reset_index()
nbr_marie = nbr_marie[nbr_marie['preusuel'] == 'MARIE']
nbr_marie

annais preusuel  nombre
657      1900    MARIE   49752
1694     1901    MARIE   53176
2753     1902    MARIE   52889
3813     1903    MARIE   51444
4882     1904    MARIE   51075
...       ...      ...     ...
225613   2015    MARIE    1007
230071   2016    MARIE     935
234482   2017    MARIE     839
238844   2018    MARIE     758
260447   XXXX    MARIE    2991

[120 rows x 3 columns]

 En 2017, on recensait 13.000 prénoms différents, soit 7,6 fois plus qu’en 1900.

In [13]:
len(list(df['preusuel'][df['annais'] == '1900'].unique()))

998

In [14]:
len(list(df['preusuel'][df['annais'] == '2017'].unique()))

4399

In [15]:
4398 / 997

4.41123370110331

Faux en 2017 on recensée 4399 prenoms différents soit 4,4 fois plus qu'en 1900

 “ En 1900, parmi les prénoms recensés par l’Insee, le plus donné, Marie,
représentait 11% des naissances.”

In [16]:
nbr_annee

annais  sexe    nombre  total_naissances
0     1900     1  0.427855            393432
1     1900     2  0.572145            393432
2     1901     1  0.432467            431732
3     1901     2  0.567533            431732
4     1902     1  0.439457            443886
..     ...   ...       ...               ...
235   2017     2  0.473617            573677
236   2018     1  0.527286            564259
237   2018     2  0.472714            564259
238   XXXX     1  0.446006           8261688
239   XXXX     2  0.553994           8261688

[240 rows x 4 columns]

In [17]:
nbr_marie

annais preusuel  nombre
657      1900    MARIE   49752
1694     1901    MARIE   53176
2753     1902    MARIE   52889
3813     1903    MARIE   51444
4882     1904    MARIE   51075
...       ...      ...     ...
225613   2015    MARIE    1007
230071   2016    MARIE     935
234482   2017    MARIE     839
238844   2018    MARIE     758
260447   XXXX    MARIE    2991

[120 rows x 3 columns]

In [18]:
49752 / 393432

0.12645641432318672

En 1900 Marie représentée 12% des prenoms

“ En 2017, des «prénoms rares» ont été donnés à près de 55 000 enfants, soit 10
fois plus que le prénom le plus donné (Gabriel). “

In [65]:
nbr_prenom = df.groupby(['annais', 'preusuel'])['nombre'].sum().reset_index()
nbr_prenom

annais  preusuel  nombre
0        1900      ABEL     382
1        1900   ABRAHAM       9
2        1900   ACHILLE     152
3        1900  ACHILLES       4
4        1900      ADAM       9
...       ...       ...     ...
274032   XXXX      ÖMER     243
274033   XXXX      ÖZGE      30
274034   XXXX     ÖZGÜR      23
274035   XXXX     ÖZKAN      22
274036   XXXX     ÖZLEM      95

[274037 rows x 3 columns]

In [20]:
nbr_prenom[nbr_prenom['preusuel'] == 'GABRIEL']

annais preusuel  nombre
431      1900  GABRIEL    1292
1444     1901  GABRIEL    1465
2509     1902  GABRIEL    1658
3565     1903  GABRIEL    1580
4629     1904  GABRIEL    1567
...       ...      ...     ...
224267   2015  GABRIEL    5646
228712   2016  GABRIEL    5870
233143   2017  GABRIEL    5431
237522   2018  GABRIEL    5418
250758   XXXX  GABRIEL    2666

[120 rows x 3 columns]

In [21]:
nbr_prenom[nbr_prenom['preusuel'] == '_PRENOMS_RARES']

annais        preusuel  nombre
997      1900  _PRENOMS_RARES    2934
2061     1901  _PRENOMS_RARES    3091
3112     1902  _PRENOMS_RARES    3055
4176     1903  _PRENOMS_RARES    3024
5235     1904  _PRENOMS_RARES    3221
...       ...             ...     ...
227286   2015  _PRENOMS_RARES   51482
231726   2016  _PRENOMS_RARES   52111
236127   2017  _PRENOMS_RARES   53190
240497   2018  _PRENOMS_RARES   53978
273878   XXXX  _PRENOMS_RARES    1705

[120 rows x 3 columns]

In [22]:
53190 / 5431

9.793776468422022

Les prenoms rares representent environ 9,8 fois le prenom Gabriel en 2017

“ Un pic a été atteint en 2012, avec plus de 13.643 prénoms recensés. ”

In [23]:
df_intermediaire = nbr_prenom.groupby('annais')['preusuel'].count().reset_index()
df_intermediaire.rename(columns={'preusuel': 'nombre_prenom'}, inplace=True)

In [24]:
df_intermediaire['annais'].replace("XXXX", "0000", inplace=True)

C:\Users\Adrien\AppData\Local\Temp\ipykernel_972\455567895.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_intermediaire['annais'].replace("XXXX", "0000", inplace=True)


In [25]:
df_intermediaire['annais'] = df_intermediaire['annais'].astype(int)

In [26]:
df_intermediaire

annais  nombre_prenom
0      1900            998
1      1901           1064
2      1902           1051
3      1903           1064
4      1904           1059
..      ...            ...
115    2015           4442
116    2016           4446
117    2017           4399
118    2018           4375
119       0          33483

[120 rows x 2 columns]

In [28]:
nbr_prenom = nbr_prenom.merge(df_intermediaire, on='annais')

ValueError: You are trying to merge on object and int64 columns for key 'annais'. If you wish to proceed you should use pd.concat

In [29]:
df_intermediaire

annais  nombre_prenom
0      1900            998
1      1901           1064
2      1902           1051
3      1903           1064
4      1904           1059
..      ...            ...
115    2015           4442
116    2016           4446
117    2017           4399
118    2018           4375
119       0          33483

[120 rows x 2 columns]

In [30]:
df['preusuel'].unique()

array(['A', 'AADAM', 'AADEL', ..., 'ÏNAYA', 'ÖZGE', 'ÖZLEM'],
      shape=(33484,), dtype=object)

En 2012 seulement 4419 prenoms était recensée le pic est en 2014 pour 4595 prenoms différents

Evolution des prénoms composés contenant Marie au cours du temps
(Marie-Pierre, Marie-Paul, ….) ?

In [31]:
nbr_prenom[nbr_prenom['preusuel'].str.match('MARIE-')]

annais          preusuel  nombre
658      1900        MARIE-ANGE       6
659      1900        MARIE-ANNE      42
660      1900  MARIE-ANTOINETTE      61
661      1900     MARIE-EUGÉNIE       3
662      1900   MARIE-FRANÇOISE      20
...       ...               ...     ...
260933   XXXX      MARIE-YVETTE     120
260934   XXXX      MARIE-YVONNE     999
260935   XXXX         MARIE-ZOÉ      44
260936   XXXX         MARIE-ÈVE      21
260937   XXXX       MARIE-ÉLISE      23

[5104 rows x 3 columns]

In [32]:
nbr_marie_ = nbr_prenom[nbr_prenom['preusuel'].str.match('MARIE-')]

In [33]:
nbr_marie_ = nbr_marie_.groupby('annais')['nombre'].sum().reset_index()

In [34]:
nbr_marie_

annais  nombre
0     1900    1610
1     1901    1776
2     1902    1820
3     1903    1766
4     1904    1662
..     ...     ...
115   2015      16
116   2016      13
117   2017      24
118   2018      18
119   XXXX  167152

[120 rows x 2 columns]

Les prenoms composée de Marie ont depuis 1900 augumenter chaque année jusu'en 1946 où le nombre a explosé passant d'environ 7000 à 17000 tout en augumentant significativement. Jusqu'à 1960 où cette valeur a recommencer à diminuer très fortement jusqu'à arriver en 2018 à son total le plus bas qui est de 18

 “ 1955-1960 : Les prénoms composés (Marie-Christine, Anne-Marie,...) prennent
leur envol” ?

In [35]:
nbr_composee = nbr_prenom[nbr_prenom['preusuel'].str.contains('-')]

In [36]:
nbr_composee = nbr_composee.groupby('annais')['nombre'].sum().reset_index()

In [66]:
nbr_composee

0        2124
1        2411
2        2477
3        2448
4        2299
        ...  
115      1130
116       878
117       814
118       753
119    486067
Name: nombre, Length: 120, dtype: int64

Idem que précedemment forte augumentation,pic d'augumentation en 1946 et jusqu'à 1955 où cette valeur commence à diminuer jusqu'à atteindre son taux le plus bas en 2018 de 753

“ 2015 : Les prénoms rares forment la première catégorie de prénoms “

In [38]:
nbr_prenom[nbr_prenom['annais'] == '2015'].groupby('annais')['nombre'].sum()

annais
2015    602227
Name: nombre, dtype: int64

In [39]:
nbr_prenom[nbr_prenom['preusuel'] == '_PRENOMS_RARES']

annais        preusuel  nombre
997      1900  _PRENOMS_RARES    2934
2061     1901  _PRENOMS_RARES    3091
3112     1902  _PRENOMS_RARES    3055
4176     1903  _PRENOMS_RARES    3024
5235     1904  _PRENOMS_RARES    3221
...       ...             ...     ...
227286   2015  _PRENOMS_RARES   51482
231726   2016  _PRENOMS_RARES   52111
236127   2017  _PRENOMS_RARES   53190
240497   2018  _PRENOMS_RARES   53978
273878   XXXX  _PRENOMS_RARES    1705

[120 rows x 3 columns]

In [40]:
51482 / 602227 * 100

8.548603765689682

Sur 602227 prenoms 51482 sont des prenoms rares ce qui represente environ 8,5%

[Plus dur] Combien de prénoms faut-il pour nommer 50% des bébés par année
(e.g. en 1900 27 prénoms suffisent) ?

In [41]:
nbr_prenom

annais  preusuel  nombre
0        1900      ABEL     382
1        1900   ABRAHAM       9
2        1900   ACHILLE     152
3        1900  ACHILLES       4
4        1900      ADAM       9
...       ...       ...     ...
274032   XXXX      ÖMER     243
274033   XXXX      ÖZGE      30
274034   XXXX     ÖZGÜR      23
274035   XXXX     ÖZKAN      22
274036   XXXX     ÖZLEM      95

[274037 rows x 3 columns]

In [42]:
nbr_prenom_prop = nbr_prenom

In [43]:
nbr_prenom_prop = nbr_prenom_prop.merge(totaux_par_annee, on='annais')

In [44]:
nbr_prenom_prop['nombre'] = (nbr_prenom_prop['nombre'] / nbr_prenom_prop['total_naissances']) * 100

In [45]:
nbr_prenom_prop

annais  preusuel    nombre  total_naissances
0        1900      ABEL  0.097094            393432
1        1900   ABRAHAM  0.002288            393432
2        1900   ACHILLE  0.038634            393432
3        1900  ACHILLES  0.001017            393432
4        1900      ADAM  0.002288            393432
...       ...       ...       ...               ...
274032   XXXX      ÖMER  0.002941           8261688
274033   XXXX      ÖZGE  0.000363           8261688
274034   XXXX     ÖZGÜR  0.000278           8261688
274035   XXXX     ÖZKAN  0.000266           8261688
274036   XXXX     ÖZLEM  0.001150           8261688

[274037 rows x 4 columns]

In [46]:
nbr_prenom_prop.groupby('annais')['nombre'].sum()

annais
1900    100.000000
1901    100.000000
1902    100.000000
1903    100.000000
1904    100.000000
           ...    
2015    100.000000
2016    100.000000
2017    100.000000
2018    100.000000
XXXX     99.999625
Name: nombre, Length: 120, dtype: float64

In [47]:
mes_colonnes = ['annais', 'nombre']

In [48]:
nbr_prenom_prop[mes_colonnes].sort_values(by=mes_colonnes)

annais    nombre
6        1900  0.000763
11       1900  0.000763
18       1900  0.000763
27       1900  0.000763
33       1900  0.000763
...       ...       ...
248578   XXXX  0.073230
242578   XXXX  0.073677
250539   XXXX  0.074646
245293   XXXX  0.089437
241874   XXXX  0.090393

[274037 rows x 2 columns]

In [55]:
nbr_prenom_prop['annais'] = nbr_prenom_prop['annais'].replace('XXXX', '0')

In [56]:
nbr_prenom_prop['annais'] = nbr_prenom_prop['annais'].astype(int)

In [57]:
nbr_prenom_prop[mes_colonnes].sort_values(by=mes_colonnes).cumsum(axis=1)

annais       nombre
271952     0.0     0.000048
252336     0.0     0.000061
266313     0.0     0.000073
250677     0.0     0.000085
252337     0.0     0.000085
...        ...          ...
238543  2018.0  2018.781733
238672  2018.0  2018.787582
239613  2018.0  2018.813279
237522  2018.0  2018.960197
240497  2018.0  2027.566174

[274037 rows x 2 columns]

In [63]:
nbr_prenom_prop.sort_values(by=['annais', 'nombre'])

annais        preusuel    nombre  total_naissances
271952       0             UTE  0.000048           8261688
252336       0           HEIDE  0.000061           8261688
266313       0       PRIMITIVE  0.000073           8261688
250677       0       FREDERICH  0.000085           8261688
252337       0      HEIDEMARIE  0.000085           8261688
...        ...             ...       ...               ...
238543    2018           LOUIS  0.781733            564259
238672    2018             LÉO  0.787582            564259
239613    2018         RAPHAËL  0.813279            564259
237522    2018         GABRIEL  0.960197            564259
240497    2018  _PRENOMS_RARES  9.566174            564259

[274037 rows x 4 columns]

In [58]:
nbr_prenom

annais  preusuel  nombre
0        1900      ABEL     382
1        1900   ABRAHAM       9
2        1900   ACHILLE     152
3        1900  ACHILLES       4
4        1900      ADAM       9
...       ...       ...     ...
274032   XXXX      ÖMER     243
274033   XXXX      ÖZGE      30
274034   XXXX     ÖZGÜR      23
274035   XXXX     ÖZKAN      22
274036   XXXX     ÖZLEM      95

[274037 rows x 3 columns]

In [82]:
v = nbr_prenom_prop[nbr_prenom_prop['annais'] == 1900]
v = nbr_prenom_prop.groupby("annais")
c = v["nombre"].cumsum()
v.join(c, rsuffix='_cum')
c

0          0.097094
1          0.099382
2          0.138016
3          0.139033
4          0.141320
            ...    
274032    99.997567
274033    99.997930
274034    99.998209
274035    99.998475
274036    99.999625
Name: nombre, Length: 274037, dtype: float64

PAS REUSSI !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! (Le formateur non plus :)

La part des 10 prénoms les plus utilisés a-t-elle bien été divisée par 5 entre 1900 et
aujourd’hui ?

In [85]:
nbr_prenom_prop.sort_values(by=['annais', 'nombre'], ascending=[True, False])

annais preusuel    nombre  total_naissances
241874       0     ALIX  0.090393           8261688
245293       0  CAMILLE  0.089437           8261688
250539       0   FRANCE  0.074646           8261688
242578       0   ANDRÉA  0.073677           8261688
248578       0     ELIE  0.073230           8261688
...        ...      ...       ...               ...
240537    2018     ÉMMA  0.000532            564259
240539    2018    ÉNAËL  0.000532            564259
240540    2018    ÉNOLA  0.000532            564259
240543    2018    ÉRINE  0.000532            564259
240552    2018    ÎMRAN  0.000532            564259

[274037 rows x 4 columns]

Prenom les plus utilisées en 1900: Marie (12), Jean(3.5), Jeanne(3.5), Louis(2.3), Marguerite(2), Pierre(1.89), Joseph(1.84), Germaine(1.77), Henri(1.75), Louise(1.70).

In [126]:
liste = ['MARIE', 'JEAN', 'JEANNE', 'LOUIS', 'MARGUERITE', 'PIERRE', 'JOSEPH', 'GERMAINE', 'HENRI', 'LOUISE']

In [132]:
nbr_prenom_prop[(nbr_prenom_prop['preusuel'].isin(liste)) & (nbr_prenom_prop['annais'] == 1900)].sort_values(by='preusuel')

annais    preusuel     nombre  total_naissances
446    1900    GERMAINE   1.774131            393432
481    1900       HENRI   1.758627            393432
535    1900        JEAN   3.583847            393432
543    1900      JEANNE   3.553600            393432
559    1900      JOSEPH   1.845046            393432
613    1900       LOUIS   2.300525            393432
615    1900      LOUISE   1.701946            393432
651    1900  MARGUERITE   2.048130            393432
657    1900       MARIE  12.645641            393432
797    1900      PIERRE   1.896389            393432

In [131]:
nbr_prenom_prop[(nbr_prenom_prop['preusuel'].isin(liste)) & (nbr_prenom_prop['annais'] == 2018)].sort_values(by='preusuel')

annais    preusuel    nombre  total_naissances
237702    2018       HENRI  0.044660            564259
237962    2018        JEAN  0.132918            564259
237969    2018      JEANNE  0.342042            564259
238030    2018      JOSEPH  0.123702            564259
238543    2018       LOUIS  0.781733            564259
238546    2018      LOUISE  0.667247            564259
238833    2018  MARGUERITE  0.004253            564259
238844    2018       MARIE  0.134335            564259
239550    2018      PIERRE  0.097119            564259

Les resultats sont très variées par exemple le prenom Marie a été divisée par 510 alors que le prenom Louise a seulement été divisée par 2.5

[Plus dur] Qu’en pensez vous? (Extrait du Figaro)

In [137]:
nbr_prenom_prop[nbr_prenom_prop['preusuel'] == 'LOÏC']

annais preusuel    nombre  total_naissances
29279     1926     LOÏC  0.000490            612190
31826     1928     LOÏC  0.000654            611635
33070     1929     LOÏC  0.000829            603067
36880     1932     LOÏC  0.000650            615741
38075     1933     LOÏC  0.003084            583748
...        ...      ...       ...               ...
225337    2015     LOÏC  0.036033            602227
229788    2016     LOÏC  0.038333            586962
234211    2017     LOÏC  0.021789            573677
238568    2018     LOÏC  0.022153            564259
258761       0     LOÏC  0.024317           8261688

[91 rows x 4 columns]

In [147]:
liste_dpt = ['22', '29', '35', '56']

In [148]:
df[(df['preusuel'] == 'LOÏC') & (df['dpt'].isin(liste_dpt))].sort_values(by=['annais', 'nombre'], ascending=[True, False])

sexe preusuel annais dpt  nombre
980088     1     LOÏC   1929  22       5
980089     1     LOÏC   1932  35       4
980090     1     LOÏC   1933  22       3
980091     1     LOÏC   1933  35       3
980095     1     LOÏC   1934  22       6
...      ...      ...    ...  ..     ...
984751     1     LOÏC   2016  56       3
984782     1     LOÏC   2017  35       4
984778     1     LOÏC   2017  22       3
984780     1     LOÏC   2017  29       3
984786     1     LOÏC   2017  56       3

[299 rows x 5 columns]

Le prenom Loïc était bien populaire entre 1975 et 2000 à l'echelle regionale, même si il decroit à partir de 1990

Ce prenom était effectivement representée uniquement en bretagne a une subtilitée près. Le departement 44 n'est officiellement pas rattaché à la Bretagne bien que son attachement culturel y est fort. On peut observer que ce prenom est domiant voir exclusif à cette region et au departement 44 jusqu'en 1944 où il devient bien plus populaire dans toute la France. Cependant il ne s'efface pas de Bretagne. Il est seulement moins representer qu'ailleurs.

[Plus dur] Qu’en pensez vous? (Extrait du Figaro)

In [163]:
liste_prenom = ['ERWAN', 'ENORA', 'MAELYS', 'WILFRIED', 'ENZO', 'NINO', 'MEDHI', 'INES']
liste_moins_courant= ['ARMEL', 'MAITE', 'GIULIA']

In [164]:
df[df['preusuel'].isin(liste_moins_courant)].sort_values(by=['annais', 'nombre'], ascending=[True, False])


sexe preusuel annais dpt  nombre
147731      1    ARMEL   1900  56       5
147732      1    ARMEL   1902  56       4
147733      1    ARMEL   1903  37       6
147734      1    ARMEL   1903  56       4
147735      1    ARMEL   1904  56       4
...       ...      ...    ...  ..     ...
2458411     2   GIULIA   2018  90       3
148312      1    ARMEL   XXXX  XX    2797
2894307     2    MAITE   XXXX  XX    1836
2458418     2   GIULIA   XXXX  XX    1042
1888645     2    ARMEL   XXXX  XX     428

[1977 rows x 5 columns]

Pour certains prenom courant, par exemple Enora sont rester majoritairement dans le 29 avant de se democratiser de même pour Medhi dans le 59. Armel resta dans le 56 jusqu'en 1962 et Maite dans le 64 et 33 jusqu'en 1958. Cependant, Giulia à toujours été repandu dans tout le pays.

Lisez cette base de données avec pandas (ou autre) et regardez là droit dans les
yeux pour faire une idée de ce qu’il y a dedans.

In [165]:
Mike_Campbell

01_prenom 02_genre        03_langage  04_fréquence
0        aaliyah        f  english (modern)           0.0
1         aapeli        m           finnish           0.0
2           aapo        m           finnish           0.0
3          aaren      m,f           english           0.0
4          aarne        m           finnish           0.0
...          ...      ...               ...           ...
11622        zvi        m            jewish           0.1
11623   zvonimir        m          croatian           0.0
11624  zvonimira        f          croatian           0.0
11625     zvonko        m          croatian           0.0
11626    zygmunt        m            polish           0.0

[11627 rows x 4 columns]

Combien de prénoms sont analysés dans cette base ?

In [167]:
Mike_Campbell.shape

(11627, 4)

11627

Donner les 10 prénoms les plus multiculturels recensés dans ce dataset.

In [190]:
Mike_Campbell['temp'] = Mike_Campbell['03_langage'].str.len()

In [195]:
Mike_Campbell.sort_values(by='temp', ascending=False)

01_prenom 02_genre                                         03_langage  \
669         anna        f  english, italian, german, dutch, scandinavian,...   
7136       maria        f  italian, portuguese, german, scandinavian, dut...   
7248     martina        f  english, german, scandinavian, dutch, italian,...   
2359      daniel        m  english, jewish, french, german, scandinavian,...   
342    alexandra        f  german, scandinavian, dutch, english, greek, p...   
...          ...      ...                                                ...   
11540      zenia        f                                                NaN   
11547   zephania        m                                                NaN   
11550    zephyra        f                                                NaN   
11553       zeta        f                                                NaN   
11590       zona        f                                                NaN   

       04_fréquence   temp  
669           14.94  109.0  
7136          14.91  109.0  
7248           0.00  106.0  
2359          10.99  101.0  
342            2.26   98.0  
...             ...    ...  
11540          0.00    NaN  
11547          0.00    NaN  
11550          0.00    NaN  
11553          0.03    NaN  
11590          0.64    NaN  

[11627 rows x 5 columns]

Anna, Maria, Martina, Daniel, Alexandra, Tatiana, Silvia, Viktor, Eva, Antonia sont les 10 prenoms les plus multiculturels.

Reprenez votre dataframe des prénoms et essayez de rajouter une colonne ‘origine’
dans votre dataframe des prénoms. Essayez de le faire de deux manières : en
utilisant un mapping avec dictionnaire (voir ressource) et avec la fonction merge de
pandas.

1

In [206]:
Mike_Campbell['01_prenom'] = Mike_Campbell['01_prenom'].str.upper()

In [213]:
Mike_Campbell

01_prenom 02_genre        03_langage  04_fréquence  temp
0        AALIYAH        f  english (modern)           0.0  16.0
1         AAPELI        m           finnish           0.0   7.0
2           AAPO        m           finnish           0.0   7.0
3          AAREN      m,f           english           0.0   7.0
4          AARNE        m           finnish           0.0   7.0
...          ...      ...               ...           ...   ...
11622        ZVI        m            jewish           0.1   6.0
11623   ZVONIMIR        m          croatian           0.0   8.0
11624  ZVONIMIRA        f          croatian           0.0   8.0
11625     ZVONKO        m          croatian           0.0   8.0
11626    ZYGMUNT        m            polish           0.0   6.0

[11627 rows x 5 columns]

In [230]:
intermediaire = Mike_Campbell[['01_prenom', '03_langage']]

In [236]:
intermediaire = intermediaire.rename(columns={'01_prenom': 'preusuel'})

In [237]:
intermediaire

preusuel        03_langage
0        AALIYAH  english (modern)
1         AAPELI           finnish
2           AAPO           finnish
3          AAREN           english
4          AARNE           finnish
...          ...               ...
11622        ZVI            jewish
11623   ZVONIMIR          croatian
11624  ZVONIMIRA          croatian
11625     ZVONKO          croatian
11626    ZYGMUNT            polish

[11627 rows x 2 columns]

In [241]:
df = df.merge(intermediaire, on='preusuel')

In [242]:
df = df.rename(columns={'03_langage': 'origine'})

In [245]:
df

sexe preusuel annais  dpt  nombre            origine
0           1    AARON   1962   75       3  english, biblical
1           1    AARON   1976   75       3  english, biblical
2           1    AARON   1982   75       3  english, biblical
3           1    AARON   1984   75       3  english, biblical
4           1    AARON   1985   75       5  english, biblical
...       ...      ...    ...  ...     ...                ...
2490637     2      ÉVA   2018   93       3          hungarian
2490638     2      ÉVA   2018   95       6          hungarian
2490639     2      ÉVA   2018  974       3          hungarian
2490640     2      ÉVA   XXXX   XX     587          hungarian
2490641     2    ÖZLEM   XXXX   XX      95            turkish

[2490642 rows x 6 columns]

2

In [249]:
df = df.drop('origine', axis=1)

In [254]:
df['origine'] = df['preusuel'].map(intermediaire.set_index('preusuel')['03_langage'])

In [262]:
df

sexe preusuel annais  dpt  nombre            origine
0           1    AARON   1962   75       3  english, biblical
1           1    AARON   1976   75       3  english, biblical
2           1    AARON   1982   75       3  english, biblical
3           1    AARON   1984   75       3  english, biblical
4           1    AARON   1985   75       5  english, biblical
...       ...      ...    ...  ...     ...                ...
2490637     2      ÉVA   2018   93       3          hungarian
2490638     2      ÉVA   2018   95       6          hungarian
2490639     2      ÉVA   2018  974       3          hungarian
2490640     2      ÉVA   XXXX   XX     587          hungarian
2490641     2    ÖZLEM   XXXX   XX      95            turkish

[2490642 rows x 6 columns]

Quelle est la représentativité de cette base de données ? Quel pourcentage de
prénoms arrive-t-on à faire correspondre à la base de données précédente ?

In [265]:
df_copy = pd.read_csv(r"C:\Users\Adrien\Documents\datascience\Projet Prenom\dpt2018.csv", sep=';')

In [264]:
df.shape

(2490642, 6)

In [267]:
df_copy.shape

(3624994, 5)

In [269]:
2490642 / 3624994

0.6870747923996564

Environ 68% de ces base de données corespondent

Essayer de retrouver certaines tendances temporelles dans l’attribution des
prénoms ? Donne-t-on plus de prénoms d’origine différente maintenant ? Quand
est-ce que ça s’est diversifié ? Y a-t-il des origines qui ont disparu ? Qu’en est-il des
prénoms Basques ? Bretons ? Catalans ?

In [283]:
df['temp'] = df['origine'].str.len()

In [284]:
df.sort_values(by=['annais', 'temp'], ascending=[True, False])

sexe  preusuel annais dpt  nombre  \
1276596     2      ANNA   1900  01      12   
1276597     2      ANNA   1900  02       7   
1276598     2      ANNA   1900  03      19   
1276599     2      ANNA   1900  04       7   
1276600     2      ANNA   1900  05       5   
...       ...       ...    ...  ..     ...   
2451264     2      VINA   XXXX  XX      37   
2463065     2  WALBURGA   XXXX  XX      22   
2465025     2     WILDA   XXXX  XX      42   
2465381     2       YAN   XXXX  XX      26   
2485699     2     ZENIA   XXXX  XX      96   

                                                   origine   temp  
1276596  english, italian, german, dutch, scandinavian,...  109.0  
1276597  english, italian, german, dutch, scandinavian,...  109.0  
1276598  english, italian, german, dutch, scandinavian,...  109.0  
1276599  english, italian, german, dutch, scandinavian,...  109.0  
1276600  english, italian, german, dutch, scandinavian,...  109.0  
...                                                    ...    ...  
2451264                                                NaN    NaN  
2463065                                                NaN    NaN  
2465025                                                NaN    NaN  
2465381                                                NaN    NaN  
2485699                                                NaN    NaN  

[2490642 rows x 7 columns]